python_modules/dagit/dagit_tests/test_graphql.py

```diff
@@ -1982,8 +1735,61 @@ def test_successful_start_subplan():
             with open(out_df_file, 'rb') as ff:
                 out_df = pickle.load(ff)
             assert out_df.to_dict('list') == {'num1': [1, 3], 'num2': [2, 4], 'sum': [3, 7]}
-            if result.errors:
-                raise Exception(result.errors)
+
+    if result.errors:
+        raise Exception(result.errors)
+
+    query_result = result.data['startSubplanExecution']
+
+    assert query_result['__typename'] == 'StartSubplanExecutionSuccess'
+    assert query_result['pipeline']['name'] == 'pandas_hello_world'
+    assert query_result['hasFailures'] is False
+    step_results = {
+        step_result['step']['key']: step_result for step_result in query_result['stepResults']
+    }
+
+    assert 'sum_solid.transform' in step_results
+
+    assert step_results['sum_solid.transform']['__typename'] == 'StepSuccessResult'
+    assert step_results['sum_solid.transform']['success'] is True
+    assert step_results['sum_solid.transform']['outputName'] == 'result'
+    assert (
+        step_results['sum_solid.transform']['valueRepr']
+        == '''   num1  num2  sum
+0     1     2    3
+1     3     4    7'''
+    )
+
+
+def test_user_error_pipeline():
+    result = execute_dagster_graphql(
+        define_context(),
+        START_EXECUTION_PLAN_QUERY,
+        variables={
+            'pipelineName': 'naughty_programmer_pipeline',
+            'config': {},
+            'stepExecutions': [{'stepKey': 'throw_a_thing.transform'}],
+            'executionMetadata': {'runId': 'dljkfdlkfld'},
+        },
+    )
+
+    if result.errors:
+        raise Exception(result.errors)
+
+    assert result.data
+
+    query_result = result.data['startSubplanExecution']
```

Comment:
i would like to see us layer snapshot tests in on top of these focused assertions